### PR TITLE
Refactor ABBYY classes

### PR DESCRIPTION
**Pull request by thatbudakguy**
Pulled this pre-work out of #1191, since that will add more classes to the `Abbyy` module and it seemed time to break things up into separate files.

This also makes some small changes to `Results` (adding a `druid` accessor and slightly refactoring the tests). As discussed in slack, it removes the `druid:` prefix from our fixtures.

### CODE REVIEW

**Review by peetucket — APPROVED** [review]

File: lib/dor/text_extraction/abbyy/results.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        # Naming convention for result XML files:
+        # ab123cd4567.xml.result.xml
+        # ab123cd4567.xml.0001.result.xml
+        RESULT_FILE_PATTERN = /(?<druid>\w+)\.xml(?:\.(?<run_index>\d+))?\.result\.xml/
+
+        def initialize(result_path:)
+          @result_path = result_path
+          druid, run_index = RESULT_FILE_PATTERN.match(File.basename(result_path)).captures
+          @druid = druid
+          @run_index = (run_index || 0).to_i
```
Comment (dnoneill) [inline lib/dor/text_extraction/abbyy/results.rb:19]: Is this used for anything? Is there a reason it is being set?

Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/results.rb:19]: Nope. I wasn't sure if later we would need to differentiate between runs, and it seemed easy to do, so I set it 🤷🏻

Comment (thatbudakguy) [reply lib/dor/text_extraction/abbyy/results.rb:19]: The druid will get used by the (upcoming) `Abbyy::FileWatcher`, though.

Comment (dnoneill) [reply lib/dor/text_extraction/abbyy/results.rb:19]: Gotcha.

**Review by dnoneill — APPROVED** [review]

**Pull request merged**